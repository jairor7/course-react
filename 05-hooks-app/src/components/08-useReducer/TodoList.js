import React from "react";
import PropTypes from "prop-types";
import TodoItem from "./TodoItem";

const TodoList = ({
  todos,
  handleComplete,
  handleRemoveTodo,
}) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, index) => (
        <TodoItem
          key={todo.id}
          todo={todo}
          index={index}
          handleComplete={handleComplete}
          handleRemoveTodo={handleRemoveTodo}
        />
      ))}
    </ul>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleComplete: PropTypes.func.isRequired,
  handleRemoveTodo: PropTypes.func.isRequired,
};

export default TodoList;
