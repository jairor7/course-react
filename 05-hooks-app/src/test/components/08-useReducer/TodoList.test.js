import TodoList from "../../../components/08-useReducer/TodoList";
import { shallow } from "enzyme";
import { demoInitialStateTodos } from "../../fixtures/demoTodos";

describe("Pruebas en el componente <TodoList />", () => {
  const handleComplete = jest.fn();
  const handleRemoveTodo = jest.fn();
  const wrapper = shallow(
    <TodoList
      todos={demoInitialStateTodos}
      handleComplete={handleComplete}
      handleRemoveTodo={handleRemoveTodo}
    />
  );
  test("Debe mostrarse correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener dos <TodoItem />", () => {
    expect(wrapper.find("TodoItem").length).toBe(
      demoInitialStateTodos.length
    );
    expect(
      wrapper
        .find("TodoItem")
        .at(0)
        .prop("handleRemoveTodo")
    ).toEqual(expect.any(Function));
  });
});
