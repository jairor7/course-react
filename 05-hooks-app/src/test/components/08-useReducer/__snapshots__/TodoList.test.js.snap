// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <TodoList /> Debe mostrarse correctamente el componente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoItem
    handleComplete={[MockFunction]}
    handleRemoveTodo={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "description": "Hacer la compra",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoItem
    handleComplete={[MockFunction]}
    handleRemoveTodo={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "description": "Lavar la loza",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
